indexing-service/src/main/java/io/druid/indexing/overlord/ThreadPoolTaskRunner.java

```diff
@@ -70,21 +72,34 @@ public ThreadPoolTaskRunner(
   )
   {
     this.toolboxFactory = Preconditions.checkNotNull(toolboxFactory, "toolboxFactory");
-    this.exec = MoreExecutors.listeningDecorator(Execs.singleThreaded("task-runner-%d"));
     this.conglomerate = conglomerate;
+    for (Task.Priority priority : Task.Priority.values()) {
+      this.exec.put(
```

Comment:
Every call to the run() method uses the future result. The only one that doesn't kick the future back is `WorkerTaskMonitor`, which instead has the call to run().get() ITSELF wrapped in a multi-threaded executor.